src/particles/ImpactXParticleContainer.cpp

```diff
@@ -82,4 +82,40 @@ namespace impactx
 //        Redistribute();
 
     }
+
+    void
+    ImpactXParticleContainer::MinAndMaxPositions (
+        amrex::ParticleReal& x_min, amrex::ParticleReal& x_max,
+        amrex::ParticleReal& y_min, amrex::ParticleReal& y_max,
+        amrex::ParticleReal& z_min, amrex::ParticleReal& z_max )
+    {
+        using PType = ImpactXParticleContainer::SuperParticleType;
+
+        amrex::ReduceOps<amrex::ReduceOpMin, amrex::ReduceOpMin, amrex::ReduceOpMin, amrex::ReduceOpMax, amrex::ReduceOpMax, amrex::ReduceOpMax> reduce_ops;
+        auto r = amrex::ParticleReduce<amrex::ReduceData<amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal>>(
+            *this,
+            [=] AMREX_GPU_DEVICE(const PType& p) noexcept -> amrex::GpuTuple<amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal>
+            {
+                amrex::ParticleReal x = p.pos(0);
+                amrex::ParticleReal y = p.pos(1);
+                amrex::ParticleReal z = p.pos(2);
+                return {x, y, z, x, y, z};
+            },
+            reduce_ops);
+
+        x_min = amrex::get<0>(r);
+        y_min = amrex::get<1>(r);
+        z_min = amrex::get<2>(r);
+        x_max = amrex::get<3>(r);
+        y_max = amrex::get<4>(r);
+        z_max = amrex::get<5>(r);
+
+        amrex::ParallelDescriptor::ReduceRealMin(x_min);
```

Comment:
I think we can do this in a single MPI call instead of a sequential chain of 6 of them.
Similar to:
https://github.com/ECP-WarpX/WarpX/blob/583c78e563f7a74e89fb49ff77c7a93c0baaad78/Source/Diagnostics/ReducedDiags/BeamRelevant.cpp#L324-L340

Reply:
Still todo from previews review round :)